packages/editor-ui/src/components/TextWithHighlights.vue

```diff
@@ -0,0 +1,39 @@
+<script lang="ts" setup>
+import type { PropType } from 'vue';
+import { GenericValue } from 'n8n-workflow';
+
+const props = defineProps({
+	content: {
+		type: [Object, String, Number] as PropType<GenericValue>,
+	},
+	search: {
+		type: String,
+	},
+});
+
+const splitTextBySearch = (text = '', search = ''): { tag: 'span' | 'mark'; content: string }[] => {
+	if (!search) {
+		return [
+			{
+				tag: 'span',
+				content: text,
+			},
+		];
+	}
+
+	const pattern = new RegExp(`(${search})`, 'g');
+	const splitText = text.split(new RegExp(pattern, 'gi'));
+
+	return splitText.map((t) => ({ tag: pattern.test(t) ? 'mark' : 'span', content: t }));
```

Comment:
Is supporting regex search a requirement? If the user is unaware of the feature, search will break when using `()^${}[]`.

Reply:
nice catch.. was

Reply:
was based on current implementation.. let me find another way

Reply:
or simply escape the characters